packages/video_player/tizen/src/video_player.cc

```diff
@@ -188,6 +189,49 @@ void VideoPlayer::Play() {
   if (ret != PLAYER_ERROR_NONE) {
     throw VideoPlayerError("player_start failed", get_error_message(ret));
   }
+
+  timer = ecore_timer_add(30, ScreenSaverBlock, NULL);
+}
+
+Eina_Bool VideoPlayer::ScreenSaverBlock(void *data) {
+  LOG_DEBUG("[VideoPlayer] Screen saver blocking.");
+
+  void *handle = dlopen("libcapi-screensaver.so", RTLD_LAZY);
+  if (!handle) {
+    LOG_INFO("[VideoPlayer] dlopen failed: %s", dlerror());
+    return ECORE_CALLBACK_CANCEL;
+  }
+
+  ScreensaverOverrideReset screensaver_override_reset =
+      reinterpret_cast<ScreensaverOverrideReset>(
+          dlsym(handle, "screensaver_override_reset"));
+  if (!screensaver_override_reset) {
+    LOG_INFO("[VideoPlayer] Symbol not found: %s", dlerror());
+    dlclose(handle);
+    return ECORE_CALLBACK_CANCEL;
+  }
+  int ret = screensaver_override_reset(false);
+  if (ret != 0) {
+    throw VideoPlayerError("screensaver_override_reset failed",
+                           get_error_message(ret));
```

Comment:
A callback should not throw an exception. The exception will crash the app if not caught properly. You may just print the error and exit the callback.

Reply:
Fixed.